scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContextVariableExtractor.java

```diff
@@ -120,6 +120,27 @@
         return variables;
     }
 
+    private Map<String, Serializable> addTaskVariablesAndBackupInheritedVariables(Map<String, Serializable> variables,
+            ImmutableMap<String, TaskVariable> taskVariables) {
+        Map<String, Serializable> taskAndBackupVariables = new HashMap<>();
```

Comment:
why variables is a Map, but taskVariables is an ImmutableMap?

Reply:
@sophiesongge taskVariables is defined like that in TaskLauncherInitializer. variables is a local map where the values are (usually) Strings, where as the values of taskVariables are TaskVariable objects, which have properties such as "inherited".